### ISSUE TITLE
Cannot replicate pressure and friction coefficients in verification case

### ISSUE THREAD

**Issue by Evan-Tzanidis** [firemodels/fds#2519]
FDS Version: 6.1.2
SVN Revision Number: 20564
Compile Date: Fri, 26 Sep 2014
Smokeview Version/Revision: n/a
Operating System: MS Windows

Hello all,

I have been trying to replicate the backward_facing_step_5 verification case and I am finding discrepancies between my predictions for pressure and friction coefficients, and those shown in the verification guide (see figures below).
![image](https://cloud.githubusercontent.com/assets/13354269/8706419/816bc960-2b27-11e5-96a0-e11369fe18c8.png)
![image](https://cloud.githubusercontent.com/assets/13354269/8706773/727b0b62-2b29-11e5-8e26-588ca0a9f557.png)

To provide you with some background, I used the most recent revision of the input file from the repository and I just added some DEVC, SLCF and BNDF (attached). The velocities and reattachment point seem ok.

Pressure coefficient:
I found that the term CHARACTERISTIC_VELOCITY=7.72 has been removed in recent revisions of the input file, which resulted in massive discrepancies in the pressure coefficients. So, I "corrected" my reference velocity by dividing my pressure coefficients by 7.72^2. This brought my predictions to the same order of magnitude as those in the verification guide, but my graph is still translated relative to yours. It seems like the reference pressure (p0) is different (Cp=(p-p0)/(0.5_rho_u0^2)). I also tried re-running the simulation with the characteristic velocity term included but I got the same predictions (top figure, orange line).

Friction coefficient:
I didn't find any way to export the friction coefficient directly, so I derived it as follows:
Cf=tau/(0.5_rho_u0^2)  
where tau is the shear stress on the wall and is derived by:
tau=nu*(du/dy)     for y=0
I measured nu (dynamic viscosity) and rho (density) on the wall, used 7.72 as the characteristic velocity and calculated the velocity gradient (du/dy) from the velocity in the first cell from the wall (u_dz/2). I have no explanation for this.

The input files and spreadsheet containing these calcs are here:
https://arup.sharefile.com/d-sb97f9323f7f444b9
Can you help?

Cheers,
Evan

**Comment by rmcdermo** [comment]
Evan,
No way to attach files in GitHub.  Please set up a Google Drive or equivalent and put a link here to a shared file.
Also, computing tau_w from resolved quantities will be wrong.  You have to output the friction velocity and get tau_w from the definition u\* = sqrt(tau_w/rho).

**Comment by rmcdermo** [comment]
Also, (having read your post more carefully) if you are interested in the backward_facing_step case, have you studied the Matlab script that is used to post-process that case.  All the details will be there.

**Comment by Evan-Tzanidis** [comment]
Thank you for the quick reply!

I tried calculating the friction coefficient from U\* but it gave me another wrong curve.
In addition, this gave me only positive values, because tau=u_^2_rho.
I will have a look at the matlab code.
you can find the attachements in here: https://arup.sharefile.com/d-sb97f9323f7f444b9

**Comment by Evan-Tzanidis** [comment]
I do not have matlab currently installed, but I read through the script and it seems like the friction coefficient is calculated in the same way: 
Cf_fds = visc*u_wall/(.5*h/nx(q))/(.5*rho*U_0^2);      
where nx(q) is the number of cells across the step and h is the height of the step (that is on line 114)

The pressure coefficient is imported and plotted on lines 181-208, but it is not processed.
Therefore the pressure coefficient calculated by the matlab code should be much higher than what is shown in the verification guide.

**Comment by Evan-Tzanidis** [comment]
I do not have matlab currently installed, but I read through the script and it seems like the friction coefficient is calculated in the same way: 
Cf_fds = visc_u_wall/(.5_h/nx(q))/(.5_rho_U_0^2);  
where nx(q) is the number of cells across the step and h is the height of the step (that is on line 114)

The pressure coefficient is imported and plotted on lines 181-208, but it is not processed.
Therefore the pressure coefficient calculated by the matlab code should be much higher than what is shown in the verification guide.

**Comment by Evan-Tzanidis** [comment]
ok, I have had some progress.
The matlab code uses a constant viscosity (visc=1.7801E-5), while I used the viscosity at the exact location where the velocity was measured (DEVC_ID='nu_T'). 
So the question becomes, why does the viscosity quadruples along the length of the channel and is this real or caused by the wall function? 
![image](https://cloud.githubusercontent.com/assets/13354269/8710630/11808b42-2b43-11e5-950f-defe158f7a0a.png)

On the front of the pressure coefficient I found that the matlab code uses the experimental data to process the fds predictions:
Cp_fds = Cp_fds + Cp(end) - Cp_fds(end);       (line 189)
I can do the same to my fds predictions, but I do not understand why.
Would it be to much trouble to ask you to explain that too?

Cheers!
Evan
